opal/mca/memory/patcher/configure.m4

```diff
@@ -45,7 +45,7 @@ AC_DEFUN([MCA_opal_memory_patcher_CONFIG],[
     memory_patcher_have___syscall_prototype=0
 
     AC_MSG_CHECKING([for __curbrk symbol])
-    AC_LINK_IFELSE([AC_LANG_PROGRAM([extern char *__curbrk;],[char *tmp = __curbrk;])],
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([extern char *__curbrk;],[if (*__curbrk) return 1;])],
```

Comment:
You're right of course -- I neglected to notice that you were linking, not running.  So your original code is fine (and you're right, let's not get into including stdio.h).

Could you try with `AC_CHECK_DECLS` and see if that works for `__curbrk`, `__mmap`, `__syscall`?